docs/content/specs-defs/includes/bicep/resource/non-functional/BCPRMNFR3.md

```diff
@@ -0,0 +1,40 @@
+---
+title: BCPRMNFR3 - Resource module structure
+url: /spec/BCPRMNFR3
+geekdocNav: true
+geekdocAlign: left
+geekdocAnchor: true
+type: posts
+tags: [
+  Class-Resource, # MULTIPLE VALUES: this can be "Class-Resource" AND/OR "Class-Pattern" AND/OR "Class-Utility"
+  Type-NonFunctional, # SINGLE VALUE: this can be "Type-Functional" OR "Type-NonFunctional"
+  Category-Naming/Composition, # SINGLE VALUE: this can be "Category-Testing" OR "Category-Telemetry" OR "Category-Contribution/Support" OR "Category-Documentation" OR "Category-CodeStyle" OR "Category-Naming/Composition" OR "Category-Inputs/Outputs" OR "Category-Release/Publishing"
+  Language-Bicep, # MULTIPLE VALUES: this can be "Language-Bicep" AND/OR "Language-Terraform"
+  Severity-MUST, # SINGLE VALUE: this can be "Severity-MUST" OR "Severity-SHOULD" OR "Severity-MAY"
+  Persona-Owner, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
+  Persona-Contributor, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
+  Lifecycle-BAU, # SINGLE VALUE: this can be "Lifecycle-Initial" OR "Lifecycle-BAU" OR "Lifecycle-EOL"
+  Validation-TBD # SINGLE VALUE: this can be "Validation-BCP/Manual" OR "Validation-BCP/CI/Informational" OR "Validation-BCP/CI/Enforced"
+]
+priority: 13010
+---
+
+#### ID: BCPRMNFR3 - Category: Composition - Implementing child resources
+
+Providers like `Microsoft.Sql/servers` may have dedicated child resources such as `Microsoft.Sql/servers/databases`. In these cases, the module **MUST** host this child resource in a subfolder named after the child resource's singular name ([ref](/Azure-Verified-Modules/specs-defs/includes/shared/pattern/non-functional/PMNFR1)), so that the path to the child resource folder is consistent with its resource type. In the given example, we would have a `database` subfolder in the `server` parent folder.
```

Comment:
```suggestion
Child resource modules **MUST** be stored in a subfolder of their parent resource module and named after the child resource's singular name ([ref](/Azure-Verified-Modules/specs-defs/includes/shared/pattern/non-functional/PMNFR1)), so that the path to the child resource folder is consistent with the hyerarchy of its resource type.
For example, `Microsoft.Sql/servers` may have dedicated child resources of type `Microsoft.Sql/servers/databases`. Hence, the sql server database child module is stored in a `database` subfolder of the `server` parent folder.
```